src/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -0,0 +1,689 @@
+        private int index = 0;
+
+        SingleUseAnnotationSequenceIterator() {
+            annotationSids = new IntList(getAnnotationSidList());
+        }
+
+        @Override
+        public boolean hasNext() {
+            return index < annotationSids.size();
+        }
+
+        @Override
+        public String next() {
+            int sid = annotationSids.get(index);
+            String annotation = getSymbol(sid);
+            if (annotation == null) {
+                throw new UnknownSymbolException(sid);
+            }
+            index++;
+            return annotation;
+        }
+
+        @Override
+        public void remove() {
+            throw new UnsupportedOperationException("This iterator does not support element removal.");
+        }
+    }
+
+    /**
+     * Gets the system symbol table for the Ion version currently active.
+     * @return a system SymbolTable.
+     */
+    private SymbolTable getSystemSymbolTable() {
+        // Note: Ion 1.1 currently proposes changes to the system symbol table. If this is finalized, then
+        // 'majorVersion' cannot be used to look up the system symbol table; both 'majorVersion' and 'minorVersion'
+        // will need to be used.
+        return SharedSymbolTable.getSystemSymbolTable(getIonMajorVersion());
+    }
+
+    /**
+     * Reset the local symbol table to the system symbol table.
+     */
+    private void resetSymbolTable() {
+        // The following line is not required for correctness, but it frees the references to the old symbols,
+        // potentially allowing them to be garbage collected.
+        Arrays.fill(symbols, 0, localSymbolMaxId + 1, null);
+        localSymbolMaxId = -1;
+    }
+
+    /**
+     * Reset the list of imported shared symbol tables.
+     */
+    private void resetImports() {
+        // Note: when support for the next version of Ion is added, conditionals on 'majorVersion' and 'minorVersion'
+        // must be added here.
+        imports = ION_1_0_IMPORTS;
+        firstLocalSymbolId = imports.getMaxId() + 1;
+    }
+
+    /**
+     * Creates a shared symbol table import, resolving it from the catalog if possible.
+     * @param name the name of the shared symbol table.
+     * @param version the version of the shared symbol table.
+     * @param maxId the max_id of the shared symbol table. This value takes precedence over the actual max_id for the
+     *              shared symbol table at the requested version.
+     */
+    private SymbolTable createImport(String name, int version, int maxId) {
+        SymbolTable shared = catalog.getTable(name, version);
+        if (shared == null) {
+            // No match. All symbol IDs that fall within this shared symbol table's range will have unknown text.
+            return new SubstituteSymbolTable(name, version, maxId);
+        } else if (shared.getMaxId() != maxId || shared.getVersion() != version) {
```

Comment:
Not sure what the actual cost or impact here, but I'm curious, couldn't the versions be different but the maxIds the same? Isn't that a different case?

Reply:
In that case, the version doesn't matter, as the two would be *required* to have the same symbols.

Reply:
Taking another look at this, I did find some missing logic related to handling of shared symbol table imports with malformed or missing max_id, version, and name. Fixed in https://github.com/amazon-ion/ion-java/pull/539